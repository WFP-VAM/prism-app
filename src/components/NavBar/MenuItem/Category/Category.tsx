import React from 'react';
import { Switch, Typography, withStyles } from '@material-ui/core';

import { ILayer } from '../../utils';

export interface ICategoryProps extends ILayer {
  classes: { [key: string]: string };
}

function Category({ classes, title, layers }: ICategoryProps) {
  return (
    <div className={classes.container}>
      <Typography variant="body2" className={classes.title}>
        {title}
      </Typography>
      <hr />
      {layers.map(({ id, title: layersTitle }) => (
        <div key={id} className={classes.listContainer}>
          <Switch
            size="small"
            color="default"
            inputProps={{ 'aria-label': 'checkbox with default color' }}
          />{' '}
          <Typography variant="body1">{layersTitle}</Typography>
        </div>
      ))}
    </div>
  );
}

const styles: any = () => ({
  container: {
    marginBottom: 16,
  },

  listContainer: {
    display: 'flex',
    marginBottom: 8,
  },

  title: {
    fontWeight: 'bold',
    textAlign: 'left',
  },
});

export default withStyles(styles)(Category);
