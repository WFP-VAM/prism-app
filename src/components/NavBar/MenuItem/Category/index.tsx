import React from 'react';
import {
  Switch,
  Typography,
  withStyles,
  createStyles,
  WithStyles,
} from '@material-ui/core';

import { LayerType } from '../../../../config/types';

function Category({ classes, title, layers }: CategoryProps) {
  return (
    <div className={classes.container}>
      <Typography variant="body2" className={classes.title}>
        {title}
      </Typography>
      <hr />
      {layers.map(({ id, title: layersTitle }) => (
        <div key={id} className={classes.listContainer}>
          <Switch
            size="small"
            color="default"
            inputProps={{ 'aria-label': 'checkbox with default color' }}
          />{' '}
          <Typography variant="body1">{layersTitle}</Typography>
        </div>
      ))}
    </div>
  );
}

const styles = () =>
  createStyles({
    container: {
      marginBottom: 16,
    },

    listContainer: {
      display: 'flex',
      marginBottom: 8,
    },

    title: {
      fontWeight: 'bold',
      textAlign: 'left',
    },
  });

export interface CategoryProps extends LayerType, WithStyles<typeof styles> {}

export default withStyles(styles)(Category);
