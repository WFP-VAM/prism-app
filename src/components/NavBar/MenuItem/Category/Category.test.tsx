import React from 'react';
import { render } from '@testing-library/react';
import Category from './Category';

const props = {
  title: 'Category 1',
  layers: [
    {
      id: 'ID',
      title: 'layer',
      serverType: 'wms',
      hasDate: false,
      opacity: 0.5,
      legendText: 'legendText',
    },
  ],
};

test('examples of some things', () => {
  const { container } = render(<Category {...props} />);
  expect(container).toMatchSnapshot();
});
