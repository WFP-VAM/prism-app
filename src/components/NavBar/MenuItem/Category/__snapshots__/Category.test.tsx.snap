// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="Category-container-1"
  >
    <mock-typography
      classname="Category-title-3"
      variant="body2"
    >
      Category 1
    </mock-typography>
    <hr />
    <div
      class="Category-listContainer-2"
    >
      <mock-switch
        color="default"
        inputprops="[object Object]"
        size="small"
      />
       
      <mock-typography
        variant="body1"
      >
        layer
      </mock-typography>
    </div>
  </div>
</div>
`;
